File: javascript/26_remove_duplicates_from_sorted_array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    var l = nums.length;
    if (nums < 2) {
        return l;
    }
    var i   = 1,
        res = 1;
    for (; i < l; i ++) {
        if (nums[i] === nums[i - 1]) {
            continue;
        }
        nums[res] = nums[i];
        res ++;
    }
};

File: python/26_remove_duplicates_from_sorted_array.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def removeDuplicates(self, nums):
        l = len(nums)
        if l < 2:
            return l

        i   = 1
        res = 1

        while i < l:
            if nums[i] == nums[i - 1]:
                i += 1
                continue
            #改变原始数组
            nums[res] = nums[i]
            i += 1
            res += 1

        return res